File: app/src/main/java/com/mhammad/movieapp/ui/MainActivity.java
package com.mhammad.movieapp.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.mhammad.movieapp.R;
import com.mhammad.movieapp.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity{
    MovieViewModel viewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        viewModel = new ViewModelProvider(this).get(MovieViewModel.class);
        binding.setViewModel(viewModel);
        binding.setLifecycleOwner(this);

    }




}

File: app/src/main/java/com/mhammad/moviesappkt/ui/MainActivity.kt
package com.mhammad.moviesappkt.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.get
import com.mhammad.moviesappkt.R
import com.mhammad.moviesappkt.databinding.ActivityMainBinding
import com.mhammad.moviesappkt.pojo.MovieModel

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        val viewModel: MovieViewModel = ViewModelProvider(this).get(MovieViewModel::class.java)
        viewModel.getMovieName().observe(this, Observer {
            binding.movieNameTV.text = it
        })
        binding.getMovieBTN.setOnClickListener(View.OnClickListener {
            viewModel.getMovieName()
        })
    }
}